### PR TITLE
Dev/addons

### PR DESCRIPTION
**Pull request by 4Giedrius**
Fixed cart components logic #2

### CODE REVIEW

**Comment by zygintas** [comment]
omg

**Comment by zygintas** [comment]
yaw

**Comment by 4Giedrius** [comment]
giphy

File: src/Coupon/CouponFreeAddon.php [hunk 1]
```diff
@@ -3,53 +3,136 @@
...
     }
 }
+
+
+
+
+
+
+
+
```
Comment (zygintas) [inline src/Coupon/CouponFreeAddon.php]: such blank

**Pull request merged**